### PR TITLE
Rationalize and apply consistent project settings

### PR DESCRIPTION
**Pull request by briandealwis**
This PR creates and applies a canonical set of project settings and preferences for JDT, PDE, and m2e. 

The PR uses the [`eclipse-settings-maven-plugin`](https://github.com/BSI-Business-Systems-Integration-AG/eclipse-settings-maven-plugin) to apply a set of settings files as packaged up in a Maven `jar` artifact.  The settings artifact project is found in `eclipse/settings/`.  This settings artifact could be published as a canonical set of settings files (e.g., to Maven Central `com.google:java-settings-eclipse`), though I've included instructions for how to apply changes locally.

The settings are based on the _GoogleStyle_ settings from GPE with a few tweaks (described in the `eclipse/settings/README.md`).  One notable change: it requires us to install the [_Google Java Format_ plugin](https://github.com/google/google-java-format/) (details in `README.md`).

I changed the repo structure to make `/plugins`, `/features` and `/third_party` to be separate module projects that pull in their child project.  This simplifies the main `/pom.xml` and also allows configuring different settings for those children (i.e., to not configure `eclipse-settings-maven-plugin` on `third_party`).

The nice thing about this approach is that it simplifies configuring new projects — just invoke the `eclipse-settings` plugin.

This is part of #109

### CODE REVIEW

**Comment by briandealwis** [comment]
Funny, the JDT compiler uses the settings and has flagged things as errors, but my Eclipse instance didn't.

**Comment by briandealwis** [comment]
The google-java-format insists on `//` comments to have a leading space, but the Eclipse `//$NON-NLS-n$` externalized-string scanner doesn't like it.

**Comment by briandealwis** [comment]
An alternative is to continue using the GPE _GoogleStyle_ formatting settings, which is what we were mostly using, and not use the `google-java-format` plugin.

File: eclipse/settings/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+This project contains the canonical set of Eclipse `.settings/`
+files for Cloud Tools for Eclipse.  These settings configure Eclipse
+to use the [Google Style for Java][google-style-java].
+
+The settings currently reflect the following choices:
+
+  - uses the [Google Java Format plugin for Eclipse](google-java-format),
+    which requires Eclipse Oxygen (4.7) or later.
+  - unnecessary declared exceptions on methods are _errors_
```
Comment (elharo) [inline eclipse/settings/README.md]: runtime too or just checked?

File: eclipse/settings/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  - potential resource leaks are _warnings_
+  - serializable classes without `serialVersionUID` are _ignored_
+  - redundant generic type argument are _errors_
+
+## Updating the Settings
+
+The settings require installing the [Google Java Format plugin for
+Eclipse][google-java-format].  Installing the format plugin requires
+downloading the [latest release][google-java-format-release] (named
+`google-java-format-eclipse-plugin_XXXX.jar`)and placing the jar
```
Comment (elharo) [inline eclipse/settings/README.md]: space after )

File: eclipse/settings/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+in your Eclipse installation's `dropins/` directory (on MacOS, in
+`Eclipse.app/Content/Eclipse/dropins/`).
+
+Updating and applying the settings files is a two-step process based
+around the [`eclipse-settings-maven-plugin`][esmp] Maven plugin.
+
+  [esmp]: https://github.com/BSI-Business-Systems-Integration-AG/eclipse-settings-maven-plugin
```
Comment (elharo) [inline eclipse/settings/README.md]: I prefer direct inline links but up to you.

File: eclipse/settings/README.md (hunk 1 above)
Comment (briandealwis) [reply eclipse/settings/README.md]: Checked exceptions only.

File: eclipse/settings/README.md (hunk 3 above)
Comment (briandealwis) [reply eclipse/settings/README.md]: Me too, but the text was becoming hard to read :-/

**Comment by elharo** [comment]
Can this be merged now?

**Comment by briandealwis** [comment]
If we decide to use the `google-java-format` plugin, then I think we need to disable the `$NON-NLS-<x>$` warning.

**Comment by briandealwis** [comment]
Changed the configuration to ignore non-externalized strings / missing `$NON-NLS-x$`.
Tweaked the `README`s slightly.
Re-applied the settings.
PTAL.

**Comment by briandealwis** [comment]
Eclipse [bug 215466](https://bugs.eclipse.org/bugs/show_bug.cgi?id=215466) requests tolerating spaces in `//$NON-NLS`.

**Comment by elharo** [comment]
Travis is failing in weird ways that might be caused by this. E.g. see https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2789

#2785 may fix this, but if not we might want to roll this back.

**Pull request merged**